### PR TITLE
fix: if you have data to prepend, request it

### PR DESCRIPTION
**Pull request by daybrush**
## Issue

#280

### CODE REVIEW

File: test/unit/Infinite.spec.js [hunk 1]
```diff
@@ -171,6 +171,66 @@ import {wait, createElement} from "./helper/TestHelper";
 				expect(this.infinite.options.append.callCount).to.be.equal(1);
 				expect(this.infinite.options.prepend.callCount).to.be.equal(useRecycle ? 7 : 3);
 			});
+			it(`should check infinite prepend for small size(cache)`, async () => {
+				this.infinite.options.append = sinon.spy();
+				this.infinite.options.prepend = sinon.spy(({cache}) => {
+					if (useRecycle) {
+						expect(cache.length).to.be.ok;
```
Comment (WoodNeck) [inline test/unit/Infinite.spec.js]: What is it "to be ok"?
Does it mean it is regarded as true? 
If it is so, I think rather `expect(sth).to.be.above(n)` or `expect(sth).is.above(n)` is more natural as it's number.

File: test/unit/Infinite.spec.js [hunk 2]
```diff
@@ -171,6 +171,66 @@ import {wait, createElement} from "./helper/TestHelper";
...
+					},
+				});
+				this.items.prepend({
+					groupKey: 0,
+					items: [],
+					outlines: {
+						start: [0],
+						end: [1000],
+					},
+				});
```
Comment (WoodNeck) [inline test/unit/Infinite.spec.js]: Can items with same `groupKey` can be prepended?

File: test/unit/Infinite.spec.js [hunk 3]
```diff
@@ -171,6 +171,66 @@ import {wait, createElement} from "./helper/TestHelper";
...
+				this.infinite.scroll(threshold + 2);
+				// useRecycle: true, request prepend
+				// useRecycle: false, request prepend
+				this.infinite.scroll(threshold);
+				// useRecycle: true, request prepend
+				// useRecycle: false, request prepend
+				this.infinite.scroll(threshold - 50);
+				// useRecycle: true, request prepend
+				// useRecycle: false, request prepend
+				this.infinite.scroll(0);
```
Comment (WoodNeck) [inline test/unit/Infinite.spec.js]: How can `useRecycle` make test results different?
Could you please explain it to me?

Comment (daybrush) [reply test/unit/Infinite.spec.js]: @WoodNeck 
The useRecycle is to be seen from startCursor to endCursor.

If the useRecycle is false, startCursor is unconditional 0 and endCursor is the end index of the list.

File: test/unit/Infinite.spec.js (hunk 2 above)
Comment (daybrush) [reply test/unit/Infinite.spec.js]: @WoodNeck 
There is no association of groupKey in the scanl of the infinite. But I'll fix it.

File: test/unit/Infinite.spec.js (hunk 1 above)
Comment (daybrush) [reply test/unit/Infinite.spec.js]: @WoodNeck i'll fix it

**Pull request merged**